guix-systole/systoleos/systoleos.scm

```diff
@@ -0,0 +1,188 @@
+  #:use-module (guix)
+  #:use-module (guix channels)
+  ;; #:use-module (guix packages)
+  #:use-module (guix gexp)
+  #:use-module (guix build utils)
+  #:use-module (guix build-system trivial)
+  #:use-module ((guix licenses)
+                #:prefix license:)
+  #:use-module (gnu packages version-control)
+  #:use-module (gnu packages vim)
+  #:use-module (gnu packages curl)
+  #:use-module (gnu packages linux)
+  #:use-module (gnu packages mtools)
+  #:use-module (gnu packages package-management)
+  #:use-module (gnu packages terminals)
+  #:use-module (gnu packages xorg)
+  #:use-module (gnu services)
+  #:use-module (gnu services base)
+  #:use-module (gnu services desktop)
+  #:use-module (gnu services xorg)
+  #:use-module (gnu services lightdm)
+  #:use-module (gnu system)
+  #:use-module (gnu system image)
+  #:use-module (gnu system install)
+  #:use-module (gnu system shadow)
+  #:use-module (gnu system file-systems)
+  #:use-module (gnu system keyboard)
+  #:use-module (gnu bootloader)
+  #:use-module (gnu bootloader grub)
+  #:use-module (gnu image)
+  #:use-module (gnu packages)
+  #:use-module (nongnu packages linux)
+  #:use-module (nongnu system linux-initrd)
+  #:use-module (guix-systole services dicomd-service)
+  #:use-module (guix-systole packages slicer))
+
+;; https://substitutes.nonguix.org/signing-key.pub
+(define %signing-key
+  (plain-file "nonguix.pub"
+   "(public-key
+ (ecc
+  (curve Ed25519)
+  (q #C1FD53E5D4CE971933EC50C9F307AE2171A2D3B52C804642A7A35F84F3A4EA98#)))"))
+
+;; The Nonguix channel is necessary for the Linux kernel with nonfree blobs,
+;; required by most commercial hardware.
+(define %channels
+  (cons* (channel
+           (name 'nonguix)
+           (url "https://gitlab.com/nonguix/nonguix")
+           ;; Enable signature verification:
+           (introduction
+            (make-channel-introduction
+             "897c1a470da759236cc11798f4e0a5f7d4d59fbc"
+             (openpgp-fingerprint
+              "2A39 3FFF 68F4 EF7A 3D29  12AF 6F51 20A0 22FB B2D5"))))
+         (channel
+           (name 'guix)
+           (url "https://codeberg.org/guix/guix-mirror.git")
+           (branch "master")
+           (introduction
+            (make-channel-introduction
+             "c91e27c60864faa229198f6f0caf620275c429a2"
+             (openpgp-fingerprint
+              "2841 9AC6 5038 7440 C7E9 2FFA 2208 D209 58C1 DEB0"))))
+         (channel
+           (name 'guix-systole)
+           (url "https://github.com/SystoleOS/guix-systole")
+           ;; (branch "dev")
+           (branch "SCRUM-126-Define-package-for-SystoleOS-including-Slicer"))
+         %default-channels))
+
+(define serialise-channels
+  (scheme-file "channels.scm"
+               #~(begin
+                   (with-output-to-file %output
+                     (lambda ()
+                       (display (sexp->string '%
+                                              (%channels)))
+                       (newline))))))
```

Comment:
@oeramo I don't understand this. Could this file not be generated by us with the sepecifications above? what's the purpose of generation during build?